### PR TITLE
🌱 log to logger from context during drain

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Apart from main functions and test code we should never use the global klog logger. 
By using the one from the context we can make sure that the log lines have the right k/v pairs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: internal/controllers/machine/machine_controller.go [hunk 1]
```diff
@@ -530,8 +529,10 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 			log.Info(fmt.Sprintf("%s pod from Node", verbStr),
 				"pod", fmt.Sprintf("%s/%s", pod.Name, pod.Namespace))
 		},
-		Out:    writer{klog.Info},
-		ErrOut: writer{klog.Error},
+		Out: writer{log.Info},
```
[the remark this answers is not in the document]
Comment (fabriziopandini) [reply internal/controllers/machine/machine_controller.go:532]: Given that it deals with users' workloads I don't mind being verbose here.
But probably we will get back to this soon with more context as soon as we consolidate log improvements actions

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

File: internal/controllers/machine/machine_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbueringer) [reply internal/controllers/machine/machine_controller.go:532]: I'm fine with keeping the PR open for a bit if we want to defer. Just noticed the usage of the global klog'er here and we should fix it at some point.

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: internal/controllers/machine/machine_controller.go [hunk 2]
```diff
@@ -530,8 +529,10 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
...
+		ErrOut: writer{func(msg string, keysAndValues ...interface{}) {
+			log.Error(nil, msg, keysAndValues)
```
[the remark this answers is not in the document]
Comment (killianmuldoon) [reply internal/controllers/machine/machine_controller.go:534]: I think this is fine as you've done it here.

**Comment by sbueringer** [comment]
@fabriziopandini Do we want to merge it?

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**